### ISSUE TITLE
Wouldn't it be nice to have a IJsonConverter loaded in DefaultContentManager Ctor

### ISSUE THREAD

**Issue by CSurieux** [OrchardCMS/Orchard#5867]
With this we would be able to serialize/deserialize properties between Parts and Records or Parts and Infosets ?
Maybe  JsonSerialize/JsonDeserialize methods in IContentmanager ?

**Comment by Piedone** [comment]
So you'd add JSON serialization methods to IContentManager so it's more easily available? The issue is that IContentManager has nothing to do with JSON de/serialization (SRP). Furthermore you can simply use LazyFields in parts if you need any special storage and retrieval logic.

**Comment by CSurieux** [comment]
Sorry but I don't understand your answer.
I just say it would ne nice to be able to say 'this prop will be json serialized before going in db...and deserialized when back.  I don't want lazyfields ?? May be an attribute...

**Comment by sfmskywalker** [comment]
Being able to serialize/deserialize would be nice, but not by adding such methods to `IContentManager` as this violates the SRP as Zoltan mentions. `IContentManager` should not be concerned with serialization of content items. Instead, it would in my opinion be better to have a dedicated class be responsible for handling that. `IContentSerializer` for example, whose `IContentSerializer.Serialize` would take in a content item and returns a JSON string representing the content item.

**Comment by CSurieux** [comment]
The important thing is that it should be handy. The place of the item code is not a problem as far as we have not to write complicated tons.

Envoyé à partir de mon sanyo phc 25

-----Message d'origine-----
De : "Sipke Schoorstra" <notifications@github.com>
Envoyé : ‎28/‎09/‎2015 16:17
À : "OrchardCMS/Orchard" <Orchard@noreply.github.com>
Cc : "CSurieux" <christian.surieux@hotmail.fr>
Objet : Re: [Orchard] Wouldn't it be nice to have a IJsonConverter loaded in DefaultContentManager Ctor (#5867)

Being able to serialize/deserialize would be nice, but not by adding such methods to IContentManager as this violates the SRP as Zoltan mentions. IContentManager should not be concerned with serialization of content items. Instead, it would in my opinion be better to have a dedicated class be responsible for handling that. IContentSerializer for example, whose IContentSerializer.Serialize would take in a content item and returns a JSON string representing the content item.

**Comment by CSurieux** [comment]
Moreover I was not speaking of full contentItem but simply Part's properties.

Envoyé à partir de mon sanyo phc 25
________________________________
De : Sipke Schoorstra<mailto:notifications@github.com>
Envoyé : ‎28/‎09/‎2015 16:17
À : OrchardCMS/Orchard<mailto:Orchard@noreply.github.com>
Cc : CSurieux<mailto:christian.surieux@hotmail.fr>
Objet : Re: [Orchard] Wouldn't it be nice to have a IJsonConverter loaded in DefaultContentManager Ctor (#5867)

Being able to serialize/deserialize would be nice, but not by adding such methods to `IContentManager` as this violates the SRP as Zoltan mentions. `IContentManager` should not be concerned with serialization of content items. Instead, it would in my opinion be better to have a dedicated class be responsible for handling that. `IContentSerializer` for example, whose `IContentSerializer.Serialize` would take in a content item and returns a JSON string representing the content item.

---

**Comment by sebastienros** [comment]
I could see the point, but you can already rely on Json.NET in your part. We do it for other types already like string[] where we save them as joined elements manually.

I better approach could be using attributes to mark the properties. We could even extend this logic to use the infoset instead of using the extension methods.

I think it's not risky to add these extension methods to use JSon.NET.

**Comment by CSurieux** [comment]
Yes.  Infoset are very similar. Using an attribute for them should also be nice.
The pb with extensions is that we need to instantiate a json converter each time..or there is a better way than I do.
If in one part you have several properties to serialize....

Envoyé à partir de mon sanyo phc 25
________________________________
De : Sébastien Ros<mailto:notifications@github.com>
Envoyé : ‎28/‎09/‎2015 18:27
À : OrchardCMS/Orchard<mailto:Orchard@noreply.github.com>
Cc : CSurieux<mailto:christian.surieux@hotmail.fr>
Objet : Re: [Orchard] Wouldn't it be nice to have a IJsonConverter loaded in DefaultContentManager Ctor (#5867)

I could see the point, but you can already rely on Json.NET in your part. We do it for other types already like string[] where we save them as joined elements manually.

I better approach could be using attributes to mark the properties. We could even extend this logic to use the infoset instead of using the extension methods.

I think it's not risky to add these extension methods to use JSon.NET.

---
